test/client/fixtures/api-test.js

```diff
@@ -11,34 +11,30 @@ QUnit.testDone(function () {
 
 module('regression');
 
-asyncTest('should prevent navigation from the about:blank page to the relative url (GH-645)', function () {
-    var iframe  = document.createElement('iframe');
-    var handler = function () {
-        var iframeHammerhead = iframe.contentWindow['%hammerhead%'];
-        var timeoutId        = null;
-        var finalize         = function () {
-            if (timeoutId)
-                clearTimeout(timeoutId);
 
+test('should prevent navigation from the about:blank page to the relative url (GH-645)', function () {
+    var iframe = document.createElement('iframe');
+
+    iframe.id = 'test' + Date.now();
+
+    return hammerhead.Promise.resolve()
+        .then(function () {
+            return new hammerhead.Promise(function (resolve) {
+                iframe.setAttribute('src', 'about:blank');
+                iframe.addEventListener('load', resolve);
+                document.body.appendChild(iframe);
+            });
+        })
+        .then(function () {
+            return new hammerhead.Promise(function (resolve) {
+                iframe.addEventListener('load', resolve);
+                setTimeout(resolve, 5000);
+
+                iframe.contentWindow['%hammerhead%'].navigateTo('/test.html');
+            });
+        })
+        .then(function (event) {
+            ok(!event, 'should prevent navigation');
```

Comment:
We don't check the event.
We check the navigation, so change to the origin condition `ok(false, 'should prevent navigation');`